import_export/mixins.py

```diff
@@ -8,51 +8,81 @@
 from .signals import post_export
 
 
-class ExportViewMixin:
+class ImportExportMixin:
     formats = base_formats.DEFAULT_FORMATS
-    form_class = ExportForm
     resource_class = None
 
-    def get_export_formats(self):
-        """
-        Returns available export formats.
-        """
-        return [f for f in self.formats if f().can_export()]
-
     def get_resource_class(self):
         if not self.resource_class:
             return modelresource_factory(self.model)
         return self.resource_class
 
-    def get_export_resource_class(self):
+    def get_resource_kwargs(self, request, *args, **kwargs):
+        return {}
+
+    def get_formats(self):
+        return [f for f in self.formats if f().can_import()]
+
+
+class BaseImportMixin(ImportExportMixin):
+    def get_import_resource_class(self):
         """
-        Returns ResourceClass to use for export.
+        Returns ResourceClass to use for import.
         """
-        return self.get_resource_class()
+        return super().get_resource_class()
 
-    def get_resource_kwargs(self, request, *args, **kwargs):
-        return {}
+    def get_import_formats(self):
+        """
+        Returns available import formats.
+        """
+        return super().get_formats()
 
-    def get_export_resource_kwargs(self, request, *args, **kwargs):
-        return self.get_resource_kwargs(request, *args, **kwargs)
+    def get_import_resource_kwargs(self, request, *args, **kwargs):
+        return super().get_resource_kwargs(request, *args, **kwargs)
 
-    def get_export_data(self, file_format, queryset, *args, **kwargs):
+
+class BaseExportMixin(ImportExportMixin):
+    model = None
+
+    def get_export_formats(self):
         """
-        Returns file_format representation for given queryset.
+        Returns available export formats.
         """
+        return super().get_formats()
+
+    def get_export_resource_class(self):
+        """
+        Returns ResourceClass to use for export.
+        """
+        return super().get_resource_class()
+
+    def get_export_resource_kwargs(self, request, *args, **kwargs):
+        return super().get_resource_kwargs(request, *args, **kwargs)
+
+    def get_data_for_export(self, request, queryset, *args, **kwargs):
         resource_class = self.get_export_resource_class()
-        data = resource_class(**self.get_export_resource_kwargs(self.request))\
+        return resource_class(**self.get_export_resource_kwargs(request, *args, **kwargs))\
             .export(queryset, *args, **kwargs)
-        export_data = file_format.export_data(data)
-        return export_data
 
-    def get_export_filename(self, file_format):
+    def get_filename(self, file_format):
         date_str = now().strftime('%Y-%m-%d')
         filename = "%s-%s.%s" % (self.model.__name__,
                                  date_str,
                                  file_format.get_extension())
         return filename
 
+
+class ExportViewMixin(BaseExportMixin):
```

Comment:
I'm not sure I understand this point.  Please can you explain further?

Generally I tried to be conservative, to avoid the risk of breaking stuff, and so it would be clearer to see what had changed.  We can refactor further if you think it is safe / appropriate.

Reply:
I meant that you could move all the methods and the `model` variable from `BaseExportMixin` which is a new class, to the  existing `ExportViewMixin` class, and all should work. But I checked and `BaseExportMixin` is the one used in the admin and `ExportViewMixin` is the one used in `ExportViewFormMixin`. As you say, it is public API so we must keep `ExportViewFormMixin` which is only used in tests. All good, I see it now thanks.